From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 26 Nov 2017 22:51:43 +0100
Subject: Remove report sending

No metrics are sent anywhere in Bromite, remove the checkbox
from first run activity.
---
 .../android/java/res/layout/fre_tosanduma.xml | 11 +------
 chrome/android/java/res/values/dimens.xml     |  1 -
 .../firstrun/ToSAndUMAFirstRunFragment.java   | 30 -------------------
 3 files changed, 1 insertion(+), 41 deletions(-)

diff --git a/chrome/android/java/res/layout/fre_tosanduma.xml b/chrome/android/java/res/layout/fre_tosanduma.xml
--- a/chrome/android/java/res/layout/fre_tosanduma.xml
+++ b/chrome/android/java/res/layout/fre_tosanduma.xml
@@ -83,16 +83,7 @@
                         android:layout_marginTop="@dimen/fre_vertical_spacing"
                         android:layout_marginBottom="@dimen/fre_tos_bottom_margin"
                         android:lineSpacingMultiplier="1.4"
-                        android:textAppearance="@style/TextAppearance.TextMedium.Primary" />
-
-                    <CheckBox
-                        android:id="@+id/send_report_checkbox"
-                        android:layout_width="wrap_content"
-                        android:layout_height="wrap_content"
-                        android:lineSpacingMultiplier="1.4"
-                        android:text="@string/fre_send_report_check"
-                        android:paddingStart="@dimen/fre_tos_checkbox_padding"
-                        android:textAppearance="@style/TextAppearance.TextMedium.Primary" />
+                        android:textAppearance="@style/TextAppearance.TextMedium.Primary" android:visibility="gone" />
                 </LinearLayout>
 
                 <include
diff --git a/chrome/android/java/res/values/dimens.xml b/chrome/android/java/res/values/dimens.xml
--- a/chrome/android/java/res/values/dimens.xml
+++ b/chrome/android/java/res/values/dimens.xml
@@ -115,7 +115,6 @@
 
     <dimen name="fre_image_bottom_margin">36dp</dimen>
     <dimen name="fre_tos_image_height">110dp</dimen>
-    <dimen name="fre_tos_checkbox_padding">12dp</dimen>
     <dimen name="fre_tos_bottom_margin">16dp</dimen>
     <dimen name="fre_loading_spinner_size">48dp</dimen>
     <dimen name="fre_bottom_loading_spinner_size">24dp</dimen>
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/ToSAndUMAFirstRunFragment.java b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/ToSAndUMAFirstRunFragment.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/ToSAndUMAFirstRunFragment.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/ToSAndUMAFirstRunFragment.java
@@ -62,7 +62,6 @@ public class ToSAndUMAFirstRunFragment
     private boolean mAllowCrashUpload;
 
     private Button mAcceptButton;
-    private CheckBox mSendReportCheckBox;
     private TextView mTosAndPrivacy;
     private View mTitle;
     private View mProgressSpinner;
@@ -89,13 +88,10 @@ public class ToSAndUMAFirstRunFragment
         mProgressSpinner = view.findViewById(R.id.progress_spinner);
         mProgressSpinner.setVisibility(View.GONE);
         mAcceptButton = (Button) view.findViewById(R.id.terms_accept);
-        mSendReportCheckBox = (CheckBox) view.findViewById(R.id.send_report_checkbox);
         mTosAndPrivacy = (TextView) view.findViewById(R.id.tos_and_privacy);
 
         // Register event listeners.
         mAcceptButton.setOnClickListener((v) -> onTosButtonClicked());
-        mSendReportCheckBox.setOnCheckedChangeListener(
-                ((compoundButton, isChecked) -> mAllowCrashUpload = isChecked));
 
         // Make TextView links clickable.
         mTosAndPrivacy.setMovementMethod(LinkMovementMethod.getInstance());
@@ -133,11 +129,6 @@ public class ToSAndUMAFirstRunFragment
         if (!isVisibleToUser) {
             // Restore original enabled & visibility states, in case the user returns to the page.
             setSpinnerVisible(false);
-        } else {
-            // On certain versions of Android, the checkbox will appear unchecked upon revisiting
-            // the page.  Force it to the end state of the drawable animation as a work around.
-            // crbug.com/666258
-            mSendReportCheckBox.jumpDrawablesToCurrentState();
         }
     }
 
@@ -164,7 +155,6 @@ public class ToSAndUMAFirstRunFragment
         assert !isWaitingForNativeAndPolicyInit();
 
         setSpinnerVisible(false);
-        mSendReportCheckBox.setChecked(mAllowCrashUpload);
     }
 
     /** Implements {@link FreUMADialogCoordinator.Listener} */
@@ -188,8 +178,6 @@ public class ToSAndUMAFirstRunFragment
                             .isUsageAndCrashReportingPermittedByPolicy();
 
         updateTosText(umaDialogMayBeShown, hasChildAccount, isMetricsReportingDisabledByPolicy);
-
-        updateReportCheckbox(umaDialogMayBeShown, isMetricsReportingDisabledByPolicy);
     }
 
     private SpanInfo buildTermsOfServiceLink() {
@@ -265,20 +253,6 @@ public class ToSAndUMAFirstRunFragment
         mTosAndPrivacy.setText(SpanApplier.applySpans(tosString, spans.toArray(new SpanInfo[0])));
     }
 
-    private void updateReportCheckbox(
-            boolean umaDialogMayBeShown, boolean isMetricsReportingDisabledByPolicy) {
-        mAllowCrashUpload = getUmaCheckBoxInitialState();
-        mSendReportCheckBox.setChecked(mAllowCrashUpload);
-
-        if (!canShowUmaCheckBox()) {
-            if (!umaDialogMayBeShown) {
-                mAllowCrashUpload = (sShowUmaCheckBoxForTesting || VersionInfo.isOfficialBuild())
-                        && !isMetricsReportingDisabledByPolicy;
-            }
-            mSendReportCheckBox.setVisibility(View.GONE);
-        }
-    }
-
     private void openUmaDialog() {
         new FreUMADialogCoordinator(requireContext(),
                 ((ModalDialogManagerHolder) getActivity()).getModalDialogManager(), this,
@@ -373,10 +347,6 @@ public class ToSAndUMAFirstRunFragment
 
         mAcceptButton.setVisibility(visibility);
         mTosAndPrivacy.setVisibility(visibility);
-        // Avoid updating visibility if the UMA check box can't be shown right now.
-        if (canShowUmaCheckBox()) {
-            mSendReportCheckBox.setVisibility(visibility);
-        }
     }
 
     protected View getToSAndPrivacyText() {
--
2.25.1
